qiskit/pulse/pulse_lib/continuous.py

```diff
@@ -0,0 +1,237 @@
+# -*- coding: utf-8 -*-
+
+# Copyright 2019, IBM.
+#
+# This source code is licensed under the Apache License, Version 2.0 found in
+# the LICENSE.txt file in the root directory of this source tree.
+
+# pylint: disable=missing-return-doc
+
+"""Module for builtin continuous pulse functions."""
+
+import functools
+from typing import Union, Tuple
+
+import numpy as np
+
+
+def constant(times: np.ndarray, amp: complex) -> np.ndarray:
+    """Continuous constant pulse.
+
+    Args:
+        times: Times to output pulse for.
+        amp: Complex pulse amplitude.
+    """
+    return np.full(len(times), amp, dtype=np.complex_)
+
+
+def zero(times: np.ndarray) -> np.ndarray:
+    """Continuous zero pulse.
+
+    Args:
+        times: Times to output pulse for.
+    """
+    return constant(times, 0)
+
+
+def square(times: np.ndarray, amp: complex, period: float, phase: float = 0) -> np.ndarray:
+    """Continuous square wave.
+
+    Args:
+        times: Times to output wave for.
+        amp: Pulse amplitude. Wave range is [-amp, amp].
+        period: Pulse period, units of dt.
+        phase: Pulse phase.
+    """
+    x = times/period+phase/np.pi
+    return amp*(2*(2*np.floor(x) - np.floor(2*x)) + 1).astype(np.complex_)
+
+
+def sawtooth(times: np.ndarray, amp: complex, period: float, phase: float = 0) -> np.ndarray:
+    """Continuous sawtooth wave.
+
+    Args:
+        times: Times to output wave for.
+        amp: Pulse amplitude. Wave range is [-amp, amp].
+        period: Pulse period, units of dt.
+        phase: Pulse phase.
+    """
+    x = times/period+phase/np.pi
+    return amp*2*(x-np.floor(1/2+x)).astype(np.complex_)
+
+
+def triangle(times: np.ndarray, amp: complex, period: float, phase: float = 0) -> np.ndarray:
+    """Continuous triangle wave.
+
+    Args:
+        times: Times to output wave for.
+        amp: Pulse amplitude. Wave range is [-amp, amp].
+        period: Pulse period, units of dt.
+        phase: Pulse phase.
+    """
+    return amp*(-2*np.abs(sawtooth(times, 1, period, (phase-np.pi/2)/2)) + 1).astype(np.complex_)
+
+
+def cos(times: np.ndarray, amp: complex, freq: float, phase: float = 0) -> np.ndarray:
+    """Continuous cosine wave.
+
+    Args:
+        times: Times to output wave for.
+        amp: Pulse amplitude.
+        freq: Pulse frequency, units of 1/dt.
+        phase: Pulse phase.
+    """
+    return amp*np.cos(2*np.pi*freq*times+phase).astype(np.complex_)
+
+
+def sin(times: np.ndarray, amp: complex, freq: float, phase: float = 0) -> np.ndarray:
+    """Continuous cosine wave.
+
+    Args:
+        times: Times to output wave for.
+        amp: Pulse amplitude.
+        freq: Pulse frequency, units of 1/dt.
+        phase: Pulse phase.
+    """
+    return amp*np.sin(2*np.pi*freq*times+phase).astype(np.complex_)
+
+
+def _zero_gaussian_at(gaussian_samples, amp: float, center: float, sigma: float,
```

Comment:
I think it would be more useful for the argument to be the fixed width of the pulse (so not zero_at an absolute time but zero_at with rspect to center)